api_db.php

```diff
@@ -64,7 +64,18 @@
 	{
 		$from = intval($_GET["from"]);
 		$until = intval($_GET["until"]);
-		$dbquery = "SELECT timestamp, type, domain, client, status, forward FROM queries WHERE timestamp >= :from AND timestamp <= :until ";
+
+		// Use table "query_storage"
+		//   - replace domain ID with domain
+		//   - replace client ID with client name
+		//   - replace forward ID with forward destination
+		$dbquery = "SELECT timestamp, type,";
+		$dbquery .= " CASE typeof(domain) WHEN 'integer' THEN (SELECT domain FROM domain_by_id d WHERE d.id = q.domain) ELSE domain END domain,";
+		$dbquery .= " CASE typeof(client) WHEN 'integer' THEN (SELECT name FROM client_by_id c WHERE c.id = q.client) ELSE client END client,";
```

Comment:
Tricky bit: it will return empty client if `name` is empty in the long-term data. This happens e.g. with some IPv6 clients. It should return IP in such case. I guess you need a nested case here.

![Bildschirmfoto zu 2022-05-29 22-08-38](https://user-images.githubusercontent.com/26622301/170889526-8fc039da-bce5-44dd-9538-f9beb91f0edc.png)

P.S. As you can see (or not see): DNSSEC data is completely missing